gcivil-nyu-org/spring2020-cs-gy-9223-class
# Grafana tests

## Grafana Tests
Tests are passing to confirm that:
- Updating the sensor name through the UI updates the panel title on Grafana.
- Updating the sensor type through the UI updates the SQL query on Grafana.
- Updating both the sensor name and type through the UI updates both the title and query on the Grafana panel. 
- New `sensor_data_exists` view returns expected outputs:  
   - Returns {"status": True} when measurements exist for the sensor
   - Returns {"status": False} when no measurements exist.
   - Returns {"status": False} if the sensor id doesn't match an existing sensor.
- `Grafana.get_dashboard_url_by_name` returns expected outputs: 
    - Returns True for a successful query (dashboard that exists) 
    - Returns False for an unsuccessful query (dashboard not found)  
- `Grafana.update_dashboard_title` works as expected: 
    - Success case
    - Update fails if the new name matches the existing name
    - Update fails if the dashboard to update is not found.
